src/journey-maps/angular/components/attribution/attribution.ts

```diff
@@ -72,18 +74,17 @@ export class SbbAttribution implements OnChanges, OnDestroy {
       return;
     }
 
-    const attributions = new Set<string>();
     const sourceCaches = this.map?.style?.sourceCaches ?? {};
 
-    for (const sourceCache of Object.values(sourceCaches)) {
-      if (sourceCache.used || sourceCache.usedForTerrain) {
-        const attribution = sourceCache.getSource().attribution;
-        this._splitAttribution(attribution).forEach((attr) => attributions.add(attr));
-      }
-    }
+    this.attributions = Object.values(sourceCaches)
+      .filter((sc) => sc.used || sc.usedForTerrain)
+      .map((sc) => sc.getSource().attribution)
+      .map((attr) => this._sanitizer.sanitize(SecurityContext.HTML, attr ?? null))
```

Comment:
should this two maps maybe be combined to avoid an extra iteration? or should the filter be before the second map so "?? null" could be removed?

Reply:
Ich persönlich bevorzuge die 2 separaten Zeile weil ich es besser lesbar finde. Bezüglich Performance meint ChatGPT 4:
"There is no significant performance advantage in combining the two adjacent map() calls into one line. JavaScript engines are highly optimized, and the performance difference between these two approaches would be negligible."

Ich kann nicht abschätzen, ob das stimmt, aber ich hätte es mal so gelassen.

Reply:
ist ok für mich, obwohl ich chat gpt nicht vollständig vertrauen würde :D. https://stackoverflow.com/questions/55680766/does-javascript-optimise-multiple-pure-filters-maps-etc